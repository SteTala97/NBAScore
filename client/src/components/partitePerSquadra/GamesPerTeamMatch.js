import React from 'react'

import TableCell from '@mui/material/TableCell';
import TableRow from '@mui/material/TableRow';
import LinearProgress from '@mui/material/LinearProgress';
import Box from '@mui/material/Box';
import Info from '@mui/icons-material/Info';
import Button from '@mui/material/Button';


function Progress(props) {
    const type = props.typeGame;
    if(type === "In Play"){
        return <LinearProgress color="success" />
    }else{
        return ""
    }
}


export default class GamesPerTeamMatch extends React.Component {

    constructor(props) {

        super(props);
    
        this.state = {
            id:                null,
            typeGame:            "",
            currentPeriods:    null,
            totalPeriods:      null,
            linescoreHome:     null,
            linescoreVisitors: null,
            teamScoreHome:     null,
            teamScoreVisitors: null,
            teamNameHome:      null,
            teamLogoHome:      null,
            teamNameVisitors:  null,
            teamLogoVisitors:  null,
            date:              null
        }
    }

    render() {
        return (
            <TableRow
                sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
                <TableCell id="content-table-type" component="th" scope="row" align="center" sx={{width:'10%'}}>
                    {this.props.data === null ? this.state.typeGame : this.props.data.typeGame}
                    <Box sx={{ width: '10vh' }}>
                    <Progress typeGame = {this.props.data===null ? this.state.typeGame : this.props.data.typeGame}></Progress>
                    </Box>
                </TableCell>
                <TableCell id="content-table-logo" align="right" sx={{width:'15%'}}>
                    <img src={this.props.data===null ? this.state.teamLogoHome : this.props.data.teamLogoHome} className="team-logo selectable-team" alt="Logo Team" onClick={() => { this.props.setTeamClicked(true); this.props.setTeamName(this.props.data.teamNameHome) }}/>
                    <br></br><br></br>
                    <img src={this.props.data===null ? this.state.teamLogoVisitors : this.props.data.teamLogoVisitors} className="team-logo selectable-team" alt="Logo Team" onClick={() => { this.props.setTeamClicked(true); this.props.setTeamName(this.props.data.teamNameVisitors) }}/>
                </TableCell>
                <TableCell id="content-table-name" align="left" sx={{width:'15%'}}>
                    <div id="content-team-name" className='team-text selectable-team' onClick={() => { this.props.setTeamClicked(true); this.props.setTeamName(this.props.data.teamNameHome) }}> 
                        {this.props.data===null ? this.state.teamNameHome : this.props.data.teamNameHome}
                    </div>
                    <br></br><br></br><br></br>
                    <div id="content-team-name" className='team-text selectable-team' onClick={() => { this.props.setTeamClicked(true); this.props.setTeamName(this.props.data.teamNameVisitors) }}> 
                        {this.props.data===null ? this.state.teamNameVisitors : this.props.data.teamNameVisitors}
                    </div>
                </TableCell>
                <TableCell id="content-table-score" align="center" sx={{width:'10%'}}>
                    {this.props.data===null || this.props.data.teamScoreHome===null ? " - " : this.props.data.teamScoreHome}
                    <br></br><br></br><br></br>
                    {this.props.data===null || this.props.data.teamScoreVisitors===null ? " - " : this.props.data.teamScoreVisitors}
                </TableCell>
                <TableCell id="content-table-date" align="center" sx={{width:'15%'}}>{this.props.data === null ? this.state.date : this.props.data.date}</TableCell>
                <TableCell id="content-table-quarter" align="center" sx={{width:'10%'}}>{this.props.data===null ? this.state.currentPeriods : this.props.data.currentPeriods} of {this.props.data===null ? this.state.totalPeriods : this.props.data.totalPeriods}</TableCell>
                <TableCell id="content-table-pointsPerQuarter" align="center" sx={{width:'15%'}}> 
                    [
                    {this.props.data===null || this.props.data.typeGame==="Scheduled" || this.props.data.linescoreHome.length===0 ? "\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0" : this.props.data.linescoreHome.map(
                        (scorePeriod) => " "+scorePeriod+" ")}
                    ]
                    <br></br><br></br><br></br>
                    [
                    {this.props.data===null || this.props.data.typeGame==="Scheduled" || this.props.data.linescoreVisitors.length===0 ? "\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0-\xa0\xa0\xa0" : this.props.data.linescoreVisitors.map(
                        (scorePeriod) => " "+scorePeriod+" ")}
                    ]
                </TableCell>
                <TableCell id="content-table-button" align="center" sx={{width:'10%'}}> 
                    <div>
                    <Button onClick={() => { this.props.setButtonClicked(true); this.props.setGame_Id(this.props.data.id) }} variant="contained" color="success" endIcon={<Info fontSize='medium'/>} className="more-info-button"> 
                        More Info
                    </Button>
                    </div>
                </TableCell>
            </TableRow>
        );
    
    }
    
}

