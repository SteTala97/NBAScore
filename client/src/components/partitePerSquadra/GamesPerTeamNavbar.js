import React, { useEffect } from 'react';
import './GamesPerTeamNavbar.css'
const axios = require('axios');

// Selection of the team and the season for the research
const GamesPerTeamNavbar = (props) => {
    
    // When the component is loaded we require the list of the teams and the list of the seasons
    useEffect( () => {
        async function loadTeams() {
            await axios.get("https://nbascore.herokuapp.com/teams") 
                .then(function (response) {
                    props.setTeams(response.data);
                    props.setTeamError(null);
                })
                .catch(function (error) {
                    props.setTeamError(error);
                })
        }

        loadTeams();

        async function loadSeasons() {
            await axios.get("https://nbascore.herokuapp.com/seasons") 
                .then(function (response) {
                    props.setSeasons(response.data);
                    props.setSeasonError(null);
                })
                .catch(function (error) {
                    props.setSeasonError(error);
                })
        }

        loadSeasons();
    }, []);
        

    // Search the winning team
    async function search() {             
        // Missing input parameters
        if(props.team === "") {
            alert("You have to select a Team!");
        }
        else if(props.season === 0) {
            alert("You have to select a Season!");
        }
        // Call to the API needed to complete the research
        else{

            await axios.get("https://nbascore.herokuapp.com/gamesPartitePerSquadra", { params: { season: props.season, team: props.team } })
                .then(function (response) {
                    props.setGames(response.data);
                    props.setGamesError(null);
                })
                .catch(function (error) {
                    props.setGamesError(error);
                })
            props.setSearchStarted(true);
        }
    }

    // Request to obtain the list of teams has failed
    if(props.teamError){
        return (
            <div>
                Service temporarily unavailable. Unable to connect to the remote API. 
                <br />
                {props.teamError.message}
            </div>
        );
    }
    // Request to obtain the list of seasons has failed
    else if(props.seasonError){
        return (
            <div>
                Service temporarily unavailable. Unable to connect to the remote API. 
                <br />
                {props.seasonError.message}
            </div>
        );
    }
    // Everything is ok, we can render the research component
    else {
        return (
            <div className='search-menu'>
                    <form className='select'>
                        <select data-testid="teamToSearch" id="teamToSearch" onChange={e=>props.setTeam(e.target.value)}>
                            <option value="" selected disabled>Select Team</option>
                            {
                                props.teams.map((team, id) => (
                                    <option value={team.id}>{team.name}</option>
                                )) 
                            }
                        </select>
                    </form>
                    <form className='select'>
                        <select data-testid="seasonToSearch" id="seasonToSearch" onChange={e=>props.setSeason(e.target.value)}>
                            <option value="" selected disabled>Select Season</option>
                            {
                                props.seasons.map((season, id) => (
                                    <option value={season}>{season}</option>
                                )) 
                            }
                        </select>
                    </form>
            <button className="search-button" onClick={search}>Search</button> 
            </div>
        )
    };    
}

export default GamesPerTeamNavbar;
