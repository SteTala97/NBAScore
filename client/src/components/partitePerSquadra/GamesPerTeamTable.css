.GamesPerTeamTable{
  margin: 0 auto;
  margin-top: 35px;
  margin-bottom: 100px;
  }

.team-logo {
  height: 50px;
  display: block;
  margin: 0 auto;
}