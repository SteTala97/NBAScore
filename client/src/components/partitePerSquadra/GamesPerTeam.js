import React, { useState, useEffect } from 'react';
import GamesPerTeamNavbar from './GamesPerTeamNavbar';
import GamesPerTeamTable from './GamesPerTeamTable';
import GameInfo from '../visualizzazioneInformazioniPartita/GameInfo';
import TeamInfo from '../visualizzazioneInformazioniSquadra/TeamInfo';

// Root page for the visualization of the games scheduled (both finished or not) for a given team and season
function GamesPerTeam (props) {
    const [team, setTeam] = useState("");
    const [teams, setTeams] = useState([]);
    const [teamError, setTeamError] = useState(null);

    const [season, setSeason] = useState(0);
    const [seasons, setSeasons] = useState([]);
    const [seasonError, setSeasonError] = useState(null);

    const [searchStarted, setSearchStarted] = useState(false);

    const [games, setGames] = useState([]);
    const [gamesError, setGamesError] = useState(null);

    const [buttonClicked, setButtonClicked] = useState(false);
    const [gameId, setGameId] = useState("");

    useEffect(() => {
        props.setTeamClicked(false);
        props.setTeamName("");
    }, []);

    return (
        // Here there's going to be the conditional rendering to toggle between this component and GameInfo
        <React.Fragment>
            {props.teamClicked === true ?            
                <TeamInfo teamName = {props.teamName} />
                :
                buttonClicked === true ?
                    <GameInfo 
                        id = {gameId}
                    /> 
                    :
                    <React.Fragment>
                        <GamesPerTeamNavbar
                            team = {team}
                            setTeam = {setTeam}
                            teams = {teams}
                            setTeams = {setTeams}
                            teamError = {teamError}
                            setTeamError = {setTeamError}

                            season = {season}
                            setSeason = {setSeason}
                            seasons = {seasons}
                            setSeasons = {setSeasons}
                            seasonError = {seasonError}
                            setSeasonError = {setSeasonError}

                            searchStarted = {searchStarted}
                            setSearchStarted = {setSearchStarted}

                            games = {games}
                            setGames = {setGames}
                            gamesError = {gamesError}
                            setGamesError = {setGamesError}
                        />
                        <GamesPerTeamTable
                            team = {team}
                            setTeam = {setTeam}
                            teams = {teams}
                            setTeams = {setTeams}
                            teamError = {teamError}
                            setTeamError = {setTeamError}

                            season = {season}
                            setSeason = {setSeason}
                            seasons = {seasons}
                            setSeasons = {setSeasons}
                            seasonError = {seasonError}
                            setSeasonError = {setSeasonError}

                            searchStarted = {searchStarted}
                            setSearchStarted = {setSearchStarted}

                            games = {games}
                            setGames = {setGames}
                            gamesError = {gamesError}
                            setGamesError = {setGamesError}

                            setButtonClicked = {setButtonClicked}
                            setGameId = {setGameId}

                            setTeamClicked={props.setTeamClicked} 
                            setTeamName={props.setTeamName} 
                            teamClicked={props.teamClicked} 
                            teamName={props.teamName} 
                        />
                    </React.Fragment>
            }
        </React.Fragment>
    );
}

export default GamesPerTeam;
