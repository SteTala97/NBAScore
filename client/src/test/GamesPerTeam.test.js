import React from 'react';

import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';

import GamesPerTeam from '../components/partitePerSquadra/GamesPerTeam';
import GamesPerTeamMatch from '../components/partitePerSquadra/GamesPerTeamMatch';
import GamesPerTeamTable from '../components/partitePerSquadra/GamesPerTeamTable';
import GamesPerTeamNavbar from '../components/partitePerSquadra/GamesPerTeamNavbar';


const testGames = [
    { 
        id: 1,
        typeGame: 'Finished',
        currentPeriods: 4,
        totalPeriods: 4,
        linescoreHome: ["30", "20", "35", "22"],
        linescoreVisitors: ["36", "25", "36", "24"],
        teamScoreHome: 100,
        teamScoreVisitors: 110,
        teamNameHome: 'Dallas Mavericks',
        teamLogoHome: 'https://upload.wikimedia.org/wikipedia/fr/thumb/b/b8/Mavericks_de_Dallas_logo.svg/150px-Mavericks_de_Dallas_logo.svg.png',
        teamNameVisitors: 'Phoenix Suns',
        teamLogoVisitors: 'https://upload.wikimedia.org/wikipedia/fr/5/56/Phoenix_Suns_2013.png',
    },
    { 
        id: 2,
        typeGame: 'Scheduled',
        currentPeriods: 0,
        totalPeriods: 4,
        linescoreHome: [],
        linescoreVisitors: [],
        teamScoreHome: null,
        teamScoreVisitors: null,
        teamNameHome: 'Golden State Warriors',
        teamLogoHome: 'https://upload.wikimedia.org/wikipedia/fr/thumb/d/de/Warriors_de_Golden_State_logo.svg/1200px-Warriors_de_Golden_State_logo.svg.png',
        teamNameVisitors: 'Boston Celtics',
        teamLogoVisitors: 'https://upload.wikimedia.org/wikipedia/fr/thumb/6/65/Celtics_de_Boston_logo.svg/1024px-Celtics_de_Boston_logo.svg.png',
    },
    { 
        id: 3,
        typeGame: 'In Play',
        currentPeriods: 1,
        totalPeriods: 4,
        linescoreHome: ["15", "0", "0", "0"],
        linescoreVisitors: ["21", "0", "0", "0"],
        teamScoreHome: 15,
        teamScoreVisitors: 21,
        teamNameHome: 'Philadelphia 76ers',
        teamLogoHome: 'https://upload.wikimedia.org/wikipedia/fr/4/48/76ers_2016.png',
        teamNameVisitors: 'Miami Heat',
        teamLogoVisitors: 'https://upload.wikimedia.org/wikipedia/fr/thumb/1/1c/Miami_Heat_-_Logo.svg/1200px-Miami_Heat_-_Logo.svg.png',
    }
];

const testTeams = [ "Dallas Mavericks", "Phoenix Suns", "Golden State Warriors", "Boston Celtics", "Philadelphia 76ers", "Miami Heat"];
const testSeasons = [ "2015", "2016", "2017", "2018", "2019", "2020", "2021",];
const testTeam = "Dallas Mavericks";
const testSeason = "2021";
const testSearchStarted = true;
const testGamesError = false;
const testTeamError = false;
const testSeasonError = false;


describe("Rendering single components", () => {
  
    it("Renders GamesPerTeam component without crashing", () => {
        shallow(<GamesPerTeam />);
    });

    it("Renders GamesPerTeamMatch component without crashing for 'finished' game", () => {
        shallow(<GamesPerTeamMatch data={testGames[0]} />);
    });
    it("Renders GamesPerTeamMatch component without crashing for 'scheduled' game", () => {
        shallow(<GamesPerTeamMatch data={testGames[1]} />);
    });
    it("Renders GamesPerTeamMatch component without crashing for 'in play' game", () => {
        shallow(<GamesPerTeamMatch data={testGames[2]} />);
    });

    it("Renders GamesPerTeamTable component without crashing", () => {
        shallow(<GamesPerTeamTable 
                    teams={testTeams} 
                    seasons={testSeasons} 
                    searchStarted={testSearchStarted} 
                    gamesError={testGamesError} 
                    games={testGames}
                />);
    });

    it("Renders GamesPerTeamNavbar component without crashing", () => {
        shallow(<GamesPerTeamNavbar 
                    teams={testTeams} 
                    seasons={testSeasons} 
                    team={testTeam} 
                    season={testSeason} 
                    teamError={testTeamError} 
                    seasonError={testSeasonError}
                />);
    });

});

// Test that the rendering does not change when the same input data is given
describe("Snapshots testing", () => {

    it("GamesPerTeam renders correctly", () => {
        const tree = shallow(<GamesPerTeam />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it("GamesPerTeamMatch renders correctly for 'finished' game", () => {
        const tree = shallow(<GamesPerTeamMatch data={testGames[0]} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    it("GamesPerTeamMatch renders correctly for 'scheduled' game", () => {
        const tree = shallow(<GamesPerTeamMatch data={testGames[1]} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });
    it("GamesPerTeamMatch renders correctly for 'in play' game", () => {
        const tree = shallow(<GamesPerTeamMatch data={testGames[2]} />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it("GamesPerTeamTable renders correctly", () => {
        const tree = shallow(<GamesPerTeamTable 
                                teams={testTeams} 
                                seasons={testSeasons} 
                                searchStarted={testSearchStarted} 
                                gamesError={testGamesError} 
                                games={testGames}
                            />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it("GamesPerTeamNavbar renders correctly", () => {
        const tree = shallow(<GamesPerTeamNavbar 
                                teams={testTeams} 
                                seasons={testSeasons} 
                                team={testTeam} 
                                season={testSeason} 
                                teamError={testTeamError} 
                                seasonError={testSeasonError}
                            />);
        expect(shallowToJson(tree)).toMatchSnapshot();
    });

})
