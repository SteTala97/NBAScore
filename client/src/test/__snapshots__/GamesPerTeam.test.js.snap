// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots testing GamesPerTeam renders correctly 1`] = `
<Fragment>
  <GamesPerTeamNavbar
    games={Array []}
    gamesError={null}
    searchStarted={false}
    season={0}
    seasonError={null}
    seasons={Array []}
    setGames={[Function]}
    setGamesError={[Function]}
    setSearchStarted={[Function]}
    setSeason={[Function]}
    setSeasonError={[Function]}
    setSeasons={[Function]}
    setTeam={[Function]}
    setTeamError={[Function]}
    setTeams={[Function]}
    team=""
    teamError={null}
    teams={Array []}
  />
  <GamesPerTeamTable
    games={Array []}
    gamesError={null}
    searchStarted={false}
    season={0}
    seasonError={null}
    seasons={Array []}
    setButtonClicked={[Function]}
    setGameId={[Function]}
    setGames={[Function]}
    setGamesError={[Function]}
    setSearchStarted={[Function]}
    setSeason={[Function]}
    setSeasonError={[Function]}
    setSeasons={[Function]}
    setTeam={[Function]}
    setTeamError={[Function]}
    setTeams={[Function]}
    team=""
    teamError={null}
    teams={Array []}
  />
</Fragment>
`;

exports[`Snapshots testing GamesPerTeamMatch renders correctly for 'finished' game 1`] = `
<ForwardRef(TableRow)
  sx={
    Object {
      "&:last-child td, &:last-child th": Object {
        "border": 0,
      },
    }
  }
>
  <ForwardRef(TableCell)
    align="center"
    component="th"
    id="content-table-type"
    scope="row"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    Finished
    <ForwardRef(Box)
      sx={
        Object {
          "width": "10vh",
        }
      }
    >
      <Progress
        typeGame="Finished"
      />
    </ForwardRef(Box)>
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="right"
    id="content-table-logo"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src="https://upload.wikimedia.org/wikipedia/fr/thumb/b/b8/Mavericks_de_Dallas_logo.svg/150px-Mavericks_de_Dallas_logo.svg.png"
    />
    <br />
    <br />
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src="https://upload.wikimedia.org/wikipedia/fr/5/56/Phoenix_Suns_2013.png"
    />
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="left"
    id="content-table-name"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <div
      className="team-text selectable-team"
      id="content-team-name"
      onClick={[Function]}
    >
      Dallas Mavericks
    </div>
    <br />
    <br />
    <br />
    <div
      className="team-text selectable-team"
      id="content-team-name"
      onClick={[Function]}
    >
      Phoenix Suns
    </div>
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-score"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    100
    <br />
    <br />
    <br />
    110
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-date"
    sx={
      Object {
        "width": "15%",
      }
    }
  />
  <ForwardRef(TableCell)
    align="center"
    id="content-table-quarter"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    4
     of 
    4
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-pointsPerQuarter"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    [
     30 
     20 
     35 
     22 
    ]
    <br />
    <br />
    <br />
    [
     36 
     25 
     36 
     24 
    ]
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-button"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    <div>
      <ForwardRef(Button)
        className="more-info-button"
        color="success"
        endIcon={
          <Memo
            fontSize="medium"
          />
        }
        onClick={[Function]}
        variant="contained"
      >
        More Info
      </ForwardRef(Button)>
    </div>
  </ForwardRef(TableCell)>
</ForwardRef(TableRow)>
`;

exports[`Snapshots testing GamesPerTeamMatch renders correctly for 'in play' game 1`] = `
<ForwardRef(TableRow)
  sx={
    Object {
      "&:last-child td, &:last-child th": Object {
        "border": 0,
      },
    }
  }
>
  <ForwardRef(TableCell)
    align="center"
    component="th"
    id="content-table-type"
    scope="row"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    In Play
    <ForwardRef(Box)
      sx={
        Object {
          "width": "10vh",
        }
      }
    >
      <Progress
        typeGame="In Play"
      />
    </ForwardRef(Box)>
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="right"
    id="content-table-logo"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src="https://upload.wikimedia.org/wikipedia/fr/4/48/76ers_2016.png"
    />
    <br />
    <br />
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src="https://upload.wikimedia.org/wikipedia/fr/thumb/1/1c/Miami_Heat_-_Logo.svg/1200px-Miami_Heat_-_Logo.svg.png"
    />
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="left"
    id="content-table-name"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <div
      className="team-text selectable-team"
      id="content-team-name"
      onClick={[Function]}
    >
      Philadelphia 76ers
    </div>
    <br />
    <br />
    <br />
    <div
      className="team-text selectable-team"
      id="content-team-name"
      onClick={[Function]}
    >
      Miami Heat
    </div>
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-score"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    15
    <br />
    <br />
    <br />
    21
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-date"
    sx={
      Object {
        "width": "15%",
      }
    }
  />
  <ForwardRef(TableCell)
    align="center"
    id="content-table-quarter"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    1
     of 
    4
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-pointsPerQuarter"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    [
     15 
     0 
     0 
     0 
    ]
    <br />
    <br />
    <br />
    [
     21 
     0 
     0 
     0 
    ]
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-button"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    <div>
      <ForwardRef(Button)
        className="more-info-button"
        color="success"
        endIcon={
          <Memo
            fontSize="medium"
          />
        }
        onClick={[Function]}
        variant="contained"
      >
        More Info
      </ForwardRef(Button)>
    </div>
  </ForwardRef(TableCell)>
</ForwardRef(TableRow)>
`;

exports[`Snapshots testing GamesPerTeamMatch renders correctly for 'scheduled' game 1`] = `
<ForwardRef(TableRow)
  sx={
    Object {
      "&:last-child td, &:last-child th": Object {
        "border": 0,
      },
    }
  }
>
  <ForwardRef(TableCell)
    align="center"
    component="th"
    id="content-table-type"
    scope="row"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    Scheduled
    <ForwardRef(Box)
      sx={
        Object {
          "width": "10vh",
        }
      }
    >
      <Progress
        typeGame="Scheduled"
      />
    </ForwardRef(Box)>
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="right"
    id="content-table-logo"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src="https://upload.wikimedia.org/wikipedia/fr/thumb/d/de/Warriors_de_Golden_State_logo.svg/1200px-Warriors_de_Golden_State_logo.svg.png"
    />
    <br />
    <br />
    <img
      alt="Logo Team"
      className="team-logo selectable-team"
      onClick={[Function]}
      src="https://upload.wikimedia.org/wikipedia/fr/thumb/6/65/Celtics_de_Boston_logo.svg/1024px-Celtics_de_Boston_logo.svg.png"
    />
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="left"
    id="content-table-name"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    <div
      className="team-text selectable-team"
      id="content-team-name"
      onClick={[Function]}
    >
      Golden State Warriors
    </div>
    <br />
    <br />
    <br />
    <div
      className="team-text selectable-team"
      id="content-team-name"
      onClick={[Function]}
    >
      Boston Celtics
    </div>
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-score"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
     - 
    <br />
    <br />
    <br />
     - 
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-date"
    sx={
      Object {
        "width": "15%",
      }
    }
  />
  <ForwardRef(TableCell)
    align="center"
    id="content-table-quarter"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    0
     of 
    4
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-pointsPerQuarter"
    sx={
      Object {
        "width": "15%",
      }
    }
  >
    [
       -   -   -   -   
    ]
    <br />
    <br />
    <br />
    [
       -   -   -   -   
    ]
  </ForwardRef(TableCell)>
  <ForwardRef(TableCell)
    align="center"
    id="content-table-button"
    sx={
      Object {
        "width": "10%",
      }
    }
  >
    <div>
      <ForwardRef(Button)
        className="more-info-button"
        color="success"
        endIcon={
          <Memo
            fontSize="medium"
          />
        }
        onClick={[Function]}
        variant="contained"
      >
        More Info
      </ForwardRef(Button)>
    </div>
  </ForwardRef(TableCell)>
</ForwardRef(TableRow)>
`;

exports[`Snapshots testing GamesPerTeamNavbar renders correctly 1`] = `
<div
  className="search-menu"
>
  <form
    className="select"
  >
    <select
      data-testid="teamToSearch"
      id="teamToSearch"
      onChange={[Function]}
    >
      <option
        disabled={true}
        selected={true}
        value=""
      >
        Select Team
      </option>
      <option />
      <option />
      <option />
      <option />
      <option />
      <option />
    </select>
  </form>
  <form
    className="select"
  >
    <select
      data-testid="seasonToSearch"
      id="seasonToSearch"
      onChange={[Function]}
    >
      <option
        disabled={true}
        selected={true}
        value=""
      >
        Select Season
      </option>
      <option
        value="2015"
      >
        2015
      </option>
      <option
        value="2016"
      >
        2016
      </option>
      <option
        value="2017"
      >
        2017
      </option>
      <option
        value="2018"
      >
        2018
      </option>
      <option
        value="2019"
      >
        2019
      </option>
      <option
        value="2020"
      >
        2020
      </option>
      <option
        value="2021"
      >
        2021
      </option>
    </select>
  </form>
  <button
    className="search-button"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;

exports[`Snapshots testing GamesPerTeamTable renders correctly 1`] = `
<ForwardRef(TableContainer)
  className="GamesPerTeamTableContainer"
>
  <ForwardRef(Table)
    aria-label="simple table"
    className="GamesPerTeamTable"
    id="table"
    sx={
      Object {
        "maxWidth": 1000,
        "minWidth": 650,
      }
    }
  >
    <ForwardRef(TableHead)>
      <ForwardRef(TableRow)
        key="1"
      >
        <ForwardRef(TableCell)
          align="center"
          id="title-table-type"
        >
          <strong>
            Status
          </strong>
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
          colSpan={2}
          id="title-table-teams"
        >
          <strong>
            Teams
          </strong>
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
          id="title-table-points"
        >
          <strong>
            Points
          </strong>
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
          id="title-table-time"
        >
          <strong>
            Date
          </strong>
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
          id="title-table-time"
        >
          <strong>
            Quarter
          </strong>
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
          id="title-table-linescore"
        >
          <strong>
            Points per Quarter
          </strong>
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
          id="title-table-button"
        >
               
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
    </ForwardRef(TableHead)>
    <ForwardRef(TableBody)>
      <GamesPerTeamMatch
        data={
          Object {
            "currentPeriods": 4,
            "id": 1,
            "linescoreHome": Array [
              "30",
              "20",
              "35",
              "22",
            ],
            "linescoreVisitors": Array [
              "36",
              "25",
              "36",
              "24",
            ],
            "teamLogoHome": "https://upload.wikimedia.org/wikipedia/fr/thumb/b/b8/Mavericks_de_Dallas_logo.svg/150px-Mavericks_de_Dallas_logo.svg.png",
            "teamLogoVisitors": "https://upload.wikimedia.org/wikipedia/fr/5/56/Phoenix_Suns_2013.png",
            "teamNameHome": "Dallas Mavericks",
            "teamNameVisitors": "Phoenix Suns",
            "teamScoreHome": 100,
            "teamScoreVisitors": 110,
            "totalPeriods": 4,
            "typeGame": "Finished",
          }
        }
        key="1"
      />
      <GamesPerTeamMatch
        data={
          Object {
            "currentPeriods": 0,
            "id": 2,
            "linescoreHome": Array [],
            "linescoreVisitors": Array [],
            "teamLogoHome": "https://upload.wikimedia.org/wikipedia/fr/thumb/d/de/Warriors_de_Golden_State_logo.svg/1200px-Warriors_de_Golden_State_logo.svg.png",
            "teamLogoVisitors": "https://upload.wikimedia.org/wikipedia/fr/thumb/6/65/Celtics_de_Boston_logo.svg/1024px-Celtics_de_Boston_logo.svg.png",
            "teamNameHome": "Golden State Warriors",
            "teamNameVisitors": "Boston Celtics",
            "teamScoreHome": null,
            "teamScoreVisitors": null,
            "totalPeriods": 4,
            "typeGame": "Scheduled",
          }
        }
        key="2"
      />
      <GamesPerTeamMatch
        data={
          Object {
            "currentPeriods": 1,
            "id": 3,
            "linescoreHome": Array [
              "15",
              "0",
              "0",
              "0",
            ],
            "linescoreVisitors": Array [
              "21",
              "0",
              "0",
              "0",
            ],
            "teamLogoHome": "https://upload.wikimedia.org/wikipedia/fr/4/48/76ers_2016.png",
            "teamLogoVisitors": "https://upload.wikimedia.org/wikipedia/fr/thumb/1/1c/Miami_Heat_-_Logo.svg/1200px-Miami_Heat_-_Logo.svg.png",
            "teamNameHome": "Philadelphia 76ers",
            "teamNameVisitors": "Miami Heat",
            "teamScoreHome": 15,
            "teamScoreVisitors": 21,
            "totalPeriods": 4,
            "typeGame": "In Play",
          }
        }
        key="3"
      />
      <ForwardRef(TableRow) />
    </ForwardRef(TableBody)>
  </ForwardRef(Table)>
</ForwardRef(TableContainer)>
`;
